pkg/operator/ceph/csi/config.go

```diff
@@ -0,0 +1,173 @@
+/*
+Copyright 2024 The Rook Authors. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package csi
+
+import (
+	"context"
+	"strings"
+
+	cephclient "github.com/rook/rook/pkg/daemon/ceph/client"
+
+	csiopv1a1 "github.com/ceph/ceph-csi-operator/api/v1alpha1"
+	"github.com/pkg/errors"
+	v1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func CreateUpdateClientProfileRadosNamespace(ctx context.Context, c client.Client, clusterInfo *cephclient.ClusterInfo, cephBlockPoolRadosNamespacedName types.NamespacedName, clusterID, clusterName string) error {
+
+	logger.Info("creating ceph-csi clientProfile CR for rados namespace")
+
+	csiOpClientProfile := &csiopv1a1.ClientProfile{}
+	csiOpClientProfile.Name = clusterID
+	csiOpClientProfile.Namespace = cephBlockPoolRadosNamespacedName.Namespace
+	csiOpClientProfile.Spec = csiopv1a1.ClientProfileSpec{
+		CephConnectionRef: v1.LocalObjectReference{
+			Name: clusterName,
+		},
+		Rbd: &csiopv1a1.RbdConfigSpec{
+			RadosNamespace: cephBlockPoolRadosNamespacedName.Name,
+		},
+	}
+
+	err := clusterInfo.OwnerInfo.SetOwnerReference(csiOpClientProfile)
+	if err != nil {
+		return errors.Wrapf(err, "failed to set owner reference for clientProfile CR %q for radosNs", csiOpClientProfile.Name)
```

Comment:
```suggestion
		return errors.Wrapf(err, "failed to set owner reference for clientProfile CR %q for radosNamespace", csiOpClientProfile.Name)
```
Prefer using complete name in every log message.